OSDOCS-1801: Adding new audit policy options (#36229)
openshift/openshift-docs: modules/nodes-nodes-audit-policy-custom.adoc

// Module included in the following assemblies:
//
// * security/audit-log-policy-config.adoc

[id="configuring-audit-policy-custom_{context}"]
= Configuring the audit log policy with custom rules

You can configure an audit log policy that defines custom rules. You can specify multiple groups and define which profile to use for that group.

These custom rules take precedence over the top-level profile field. The custom rules are evaluated from top to bottom, and the first that matches is applied.

.Prerequisites

* You have access to the cluster as a user with the `cluster-admin` role.

.Procedure

. Edit the `APIServer` resource:
+
[source,terminal]
----
$ oc edit apiserver cluster
----

. Add the `spec.audit.customRules` field:
+
[source,yaml]
----
  apiVersion: config.openshift.io/v1
  kind: APIServer
  metadata:
  ...
  spec:
    audit:
      customRules:                        <1>
      - group: system:authenticated:oauth
        profile: WriteRequestBodies
      - group: system:authenticated
        profile: AllRequestBodies
      profile: Default                    <2>
----

> <1> Add one or more groups and specify the desired profile to use for that group. These custom rules take precedence over the top-level profile field. The custom rules are evaluated from top to bottom, and the first that matches is applied.

Reviewer: > 🚫  desire   _verb_
Do not use. Use "want".

<2> Set to `Default`, `WriteRequestBodies`, `AllRequestBodies`, or `None`. If you do not set this top-level `audit.profile` field, it defaults to the `Default` profile.
+
[WARNING]
====
It is not recommended to disable audit logging by using the `None` profile unless you are fully aware of the risks of not logging data that can be beneficial when troubleshooting issues. If you disable audit logging and a support situation arises, you might need to enable audit logging and reproduce the issue in order to troubleshoot properly.
====

. Save the file to apply the changes.

.Verification

> * Verify that a new revision of the Kubernetes API server pods has rolled out. This will take several minutes.

Reviewer: Will running the command take several minutes, or will it take a few minutes for the new revision to appear? Also, consider simplifying the tenses.

+
[source,terminal]
----
$ oc get kubeapiserver -o=jsonpath='{range .items[0].status.conditions[?(@.type=="NodeInstallerProgressing")]}{.reason}{"\n"}{.message}{"\n"}'
----
+
Review the `NodeInstallerProgressing` status condition for the Kubernetes API server to verify that all nodes are at the latest revision. The output shows `AllNodesAtLatestRevision` upon successful update:
+
[source,terminal]
----
AllNodesAtLatestRevision
3 nodes are at revision 12 <1>
----
<1> In this example, the latest revision number is `12`.
+

> If the output shows a message similar to one of the following, this means that the update is still in progress. Wait a few minutes and try again.

Reviewer: I suggest tightening this a little, and, although my ISG search skills are failing me today, I think that "following" always needs to be paired with a noun.

```suggestion
If the output shows a message similar to one of the following messages, the update is still in progress. Wait a few minutes and try again.
```
